File: BubbleSort/BubbleSort.cpp
#include <iostream>
using namespace std;

//Main Algorithm
void BubbleSort(int iArray[], int size)
{
    for (int i = 0; i < size - 1; i++)
    {
        for (int j = 0; j < size - i - 1; j++)
        {
            if (iArray[j] > iArray[j + 1])
            {
                swap(iArray[j], iArray[j + 1]);
            }
        }
    }
}

int main()
{
    int iArray[10] = { 1,5,9,3,2,4,7,8,6,0 };
    BubbleSort(iArray, sizeof(iArray) / sizeof(int));

    for (int i = 0; i < 10; i++)
    {
        cout << iArray[i];
    }
}

File: BubbleSort/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS
{
    class Program
    {
        static void Swap(ref int a, ref int b)
        {
            int temp = a;
            a = b;
            b = temp;
        }

        static void BubbleSort(int[] iArray, int size)
        {
            for (int i = 0; i < size - 1; i++)
            {
                for (int j = 0; j < size - i - 1; j++)
                {
                    if (iArray[j] > iArray[j + 1])
                    {
                        Swap(ref iArray[j], ref iArray[j + 1]);
                    }
                }
            }
        }

        private static void Main(string[] args)
        {
            int[] iArray = { 0, 5, 43, 2, 543, 25 };
            BubbleSort(iArray, iArray.Length);
        }
    }
}
